AmplifyTests/CoreTests/AmplifyTaskTests.swift

```
@@ -66,39 +66,33 @@ class AmplifyTaskTests: XCTestCase {
     }
 #endif
 
+    /// Given: A operation with discrete steps (10)
+    /// When: Its progress sequence is iterated
+    /// Then: It reports each step up to a `fractionCompleted` value that represents 100% completion
     func testLongOperation() async throws {
-        var success = false
-        var output: String? = nil
-        var thrown: Error? = nil
-
         let request = LongOperationRequest(steps: 10, delay: 0.01)
         let longTask = await runLongOperation(request: request)
 
-        Task {
-            var progressCount = 0
-            var lastProgress: Double = 0
+        var progressCount = 0
+        var lastProgress: Double = 0
+        await longTask.progress.forEach { progress in
+            lastProgress = progress.fractionCompleted
+            progressCount += 1
+        }
 
-            await longTask.progress.forEach { progress in
-                lastProgress = progress.fractionCompleted
-                progressCount += 1
-            }
+        // The first progress report happens on `fractionCompleted` 0.0
+        XCTAssertEqual(progressCount, 11)
 
-            XCTAssertEqual(progressCount, 11)
-            XCTAssertEqual(lastProgress, 100)
```

I believe this was failing silently.